docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -27,35 +29,28 @@ I/O Managers are user-provided objects that store asset and op outputs and load
-| <PyObject object="build_output_context"/>                           | Function for directly constructing an <PyObject object="OutputContext"/>, to be passed to the <PyObject object="IOManager" method="handle_output"/> method. This is designed primarily for testing purposes. |   |
+For non-asset jobs with inputs that aren't connected to upstream outputs, see the [Unconnected Inputs](/concepts/io-management/unconnected-inputs) overview.
 
-## Overview
+---
 
-Functions decorated by <PyObject object="asset" decorator/>, <PyObject object="multi_asset" decorator/>, and <PyObject object="op" decorator/> can have parameters and return values that are loaded from and written to persistent storage. <PyObject module="dagster" object="IOManager" pluralize /> let the user control how this data is stored and how it's loaded in downstream ops and assets. For `@asset` and `@multi_asset`, the I/O manager effectively determines where the physical asset lives.
+## Basics
 
-The I/O manager APIs make it easy to separate code that's responsible for logical data transformation from code that's responsible for reading and writing the results. Software-defined assets and ops can focus on business logic, while I/O managers handle I/O. This separation makes it easier to test the business logic and run it in different environments.
-
-For non-asset jobs with inputs that aren't connected to upstream outputs, see the [Unconnected Inputs](/concepts/io-management/unconnected-inputs) overview.
+- [When to use I/O managers](#when-to-use-io-managers)
+- [Outputs and downstream inputs](#outputs-and-downstream-inputs)
+- [Built-in I/O managers](#built-in-io-managers)
+- [I/O managers are resources](#io-managers-are-resources)
 
 ### When to use I/O managers
 
-I/O managers are a powerful tool that can simplify your code and allow you to easily modify where your assets are stored. They can be useful in situations where:
-
-- You store many of your assets in the same location and follow a consistent set of rules to determine the storage path.
-- You want to swap out I/O manager implementations to change how your assets are stored in local, staging, and production environments.
-- Your asset functions load the upstream dependencies into memory in order to do the function computation.
+If you find yourself writing the same code at the start and end of each asset or op to load and store data, you may want to consider factoring that code into an I/O manager. They can be useful in situations where:
 
-If you find yourself writing the same code at the start and end of each asset or op to load and store data, you may want to consider factoring that code into an I/O manager.
+- Many assets are stored in the same location and follow a consistent set of rules to determine the storage path
+- Assets should be stored differently in local, staging, and production environments
+- Aasset functions load the upstream dependencies into memory to do the function computation
```

Comment:
```suggestion
- Asset functions load the upstream dependencies into memory to do the computation
```

I definitely wrote the first version of this bullet point but now i have no idea what I was thinking with "function computation". The basic idea I was trying to get across is that if you need to load the upstream value into memory in order to do the computation that's in the asset function, then maybe I/O managers are a good choice.